### PR TITLE
fix: undefined default search bug

### LINKED ISSUE #36: Bug: default search state is "undefined", literally
When URL = localhost:3000 || speak.exchange, then you get [this](https://imgur.com/BqcJsi2).

### PR DESCRIPTION
**Pull request by Falcasantos**
checks to see if `search` is undefined; if so, then returns a welcome message

### CODE REVIEW

File: container/Exchange.js [hunk 1]
```diff
@@ -49,11 +53,19 @@ export default () => {
 
   return (
     <Container>
-      {loading || !translation ? (
-        <Spinner />
-      ) : (
-        <Translation translationResult={translation} />
-      )}
+      {(() => {
+        let element = null
```
Comment (juliuscc) [inline container/Exchange.js]: Wouldn't it make more sense to just return in the if/else-cases?

File: container/Exchange.js [hunk 2]
```diff
@@ -49,11 +53,19 @@ export default () => {
...
+        if (search === undefined) {
```
Comment (juliuscc) [inline container/Exchange.js]: This is the same as (!loading && !translation). We could bring back the if-else 4-way path that we had before

File: container/Exchange.js [hunk 3]
```diff
@@ -32,6 +32,10 @@ export default () => {
   const [translation, setTranslation] = useState(undefined)
```
Comment (juliuscc) [inline container/Exchange.js]: I think it would be good to use null here as well.

File: container/Exchange.js [hunk 4]
```diff
@@ -32,6 +32,10 @@ export default () => {
...
 
   useEffect(() => {
+    if (search === undefined) {
```
Comment (juliuscc) [inline container/Exchange.js]: Does this also work for `localhost:3000/?search=` or would `search` be an empty string in that case? What do you think of doing this instead: 
```suggestion
    if (!search) {
```

I think we catch more cases that way.

File: container/Exchange.js (hunk 1 above)
Comment (Falcasantos) [reply container/Exchange.js]: nevermind yes!

File: container/Exchange.js (hunk 4 above)
Comment (Falcasantos) [reply container/Exchange.js]: Yes good point!

File: container/Exchange.js (hunk 3 above)
Comment (Falcasantos) [reply container/Exchange.js]: Donederydoo

File: container/Exchange.js (hunk 2 above)
Comment (Falcasantos) [reply container/Exchange.js]: Actually when you load the website with no search queries, `loading` is defaulted to `true`, hence why I had to check `search` instead; `!loading && !translation` would never equal true in the default search case of no search params

Comment (juliuscc) [reply container/Exchange.js]: Maybe loading should default to false. That would make more sense, and only set loading to true if we actually do a search

Comment (Falcasantos) [reply container/Exchange.js]: Did so in the latest commit but I still put in a `|| search === ''`, so really is there any gain in doing `(!loading && !translation) || search === ''` over `!search`?

File: container/Exchange.js (hunk 4 above)
Comment (juliuscc) [reply container/Exchange.js]: This looks like it was resolved by mistake

**Review by juliuscc — APPROVED** [review]

File: container/Exchange.js (hunk 2 above)
Comment (juliuscc) [reply container/Exchange.js]: I agree

Comment (juliuscc) [reply container/Exchange.js]: In a later PR we could refactor this into a state enum like this https://kentcdodds.com/blog/stop-using-isloading-booleans

We could also add error handling

**Pull request merged**